scripts/build_custom_TAD_genelist.py

```diff
@@ -0,0 +1,111 @@
+"""
+2016 Gregory Way
+TAD Pathways
+scripts/build_custom_TAD_genelist.py
+
+Description:
+Calls two scripts from bin/ to build genelists of all genes in TADs harboring
+significant SNPs
+
+Usage:
+Run on the command line:
+
+    python scripts/build_custom_TAD_genelist.py
+
+With the following required flags:
+
+    --snp_data_file     The file name of where SNP data is stored. SNP data is
+                        generated by `scripts/tad_util/build_snp_list.R`
+    --output_file       The name of the results file to build
+
+Output:
+Trait specific .tsv files of one column each indicating all the genes that fall
+in signal TADs
+"""
+
+import os
+import argparse
+import pandas as pd
+
+# Load Command Arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("-s", "--snp_data_file", help="Location of SNP data")
+parser.add_argument("-o", "--output_file", help="Name of the output file")
+args = parser.parse_args()
+
+
+def assign_custom_snp_to_tad(snp_signal, tad_boundary):
+    """
+    Take an input snp signal and TAD boundary coordinates to output the
+    coordinates of TAD where the SNP signal resides.
+
+    Arguments:
+    :param snp_signal: Pandas Series with SNP descriptive attributes
+    :param tad_boundary: a Pandas DataFrame of all TADs and genomic locations
+
+    Output:
+    an OrderedDict object storing SNP and TAD specific info
+    """
+    chrom = ('chrom', str(snp_signal.chrom))
+    snp_loc = ('snploc', int(snp_signal.position))
+
+    tad = tad_boundary.ix[
+        (tad_boundary.chromosome == str(chrom[1])) &
+        (tad_boundary.TAD_start <= snp_loc[1]) &
+        (tad_boundary.TAD_end > snp_loc[1])
+    ]
+
+    return tad
+
+# Load Constants
+snp_data_file = args.snp_data_file
+output_file = args.output_file
+output_nearest_gene_file = '{}_nearest_gene.tsv'.format(
+    os.path.splitext(output_file)[0])
+
+# Load data
+tad_genes_df = pd.read_table(os.path.join('index',
+                                          'GENE_index_hg19_hESC.tsv.bz2'),
+                             index_col=0)
+snp_df = pd.read_table(snp_data_file)
+
+results_df = pd.DataFrame(columns=tad_genes_df.columns.tolist() + ['group'])
```

Comment:
It would be helpful if you could add more comments for lines 73-103, right now it's kind of hard to follow